mad-dog/src/helpers.js

```diff
@@ -97,27 +100,57 @@ const addUsersWithoutProfileImageOnSignUp = async (
 async function _addUsers ({ userCount, executeAll, executeOne, existingUserIds, addedUserIds, walletIndexToUserIdMap, uploadProfilePic = true }) {
   await logOps('Add users', async () => {
     const users = genRandomUsers(userCount)
+
     await executeAll(async (libs, i) => {
       try {
-      // If user already exists, do not recreate user and use existing user
+        // If user already exists, do not recreate user and use existing user
         const existingUser = await getUser({ executeOne, walletIndex: i })
         let userId
+
         if (existingUser) {
           logger.info(`Found existing user=${existingUser.user_id}`)
           existingUserIds.push(existingUser.user_id)
           userId = existingUser.user_id
+
         } else {
-          logger.info('Creating new user...')
-          const userMetadata = users[i]
-          let newUserId
-          if (uploadProfilePic) {
-            newUserId = await uploadProfileImagesAndAddUser(libs, userMetadata, USER_PIC_PATH)
-          } else {
-            newUserId = await addUser(libs, userMetadata)
+          const createNewUser = async () => {
+            const start = Date.now()
+            logger.info(`Creating new user for index ${i}...`)
+
+            const userMetadata = users[i]
+            let newUserId
+
+            if (uploadProfilePic) {
+              let picPath
+              try {
+                // get random image from web
+                const randomImageFilePath = await getRandomImageFilePath(TEMP_IMG_STORAGE_PATH)
+                picPath = randomImageFilePath
+              } catch (e) {
+                console.log(`[_addUsers] Error fetching random image: ${e.message}`)
+                picPath = USER_PIC_PATH
+              }
+              newUserId = await uploadProfileImagesAndAddUser(libs, userMetadata, picPath)
+
+            } else {
+              newUserId = await addUser(libs, userMetadata)
+            }
+
+            logger.info(`Created new user=${newUserId} in ${Date.now() - start}ms`)
+
+            addedUserIds.push(newUserId)
+            userId = newUserId
           }
-          logger.info(`Created new user=${newUserId}`)
-          addedUserIds.push(newUserId)
-          userId = newUserId
+
+          const withTimeout = async (requestPromise, timeoutMs, timeoutMsg) => {
+            const timeoutPromise = new Promise((resolve, reject) => {
+              setTimeout(() => reject(new Error(timeoutMsg)), timeoutMs)
+            })
+            return Promise.race([requestPromise, timeoutPromise])
+          }
+
+          const createNewUserTimeoutMs = 300000 // 300sec = 5 min
+          await withTimeout(createNewUser(), createNewUserTimeoutMs, `Failed to create new user for index ${i} in ${createNewUserTimeoutMs}`)
```

Comment:
ah I see the purpose of passing in the err msg here. i'll leave it up to you decide whether or not to keep this structure. we can also log context ab this fn before this call so that the fn err msg can be generic enough